docs/configuring.md

```diff
@@ -1392,14 +1414,24 @@ This allows you to keep separate the template metadata and the template code.
 
 ### `tasks`
 
--   Format: `List[str|List[str]]`
+-   Format: `List[str|List[str]|dict]`
 -   CLI flags: N/A
 -   Default value: `[]`
 
 Commands to execute after generating or updating a project from your template.
 
 They run ordered, and with the `$STAGE=task` variable in their environment.
 
+If a `dict` is given it can contain the following items:
+
+-   **command**: The task command to run.
+-   **when** (optional): Specifies a condition that needs to hold for the task to run.
```

Comment:
thanks for this work

>Commands to execute after generating or updating a project from your template.

Because tasks are run after generating AND updating, is it possible to test this in the "when" clause ?

```yaml
- command:  git clone something in_directory/
  when : `copier copy`

- command: cd in_directory && git pull
  when:  `copier update`
```

?

Reply:
Work on supporting pre/post copy/update tasks has started in #1511 and will probably build upon this PR.